### PR TITLE
Check-in protobuf Rust files

### PR DESCRIPTION
**Pull request by eoger**
So this is blocking me from landing Viaduct on mozilla-central. `tools/protobuf-gen/README.md` in this diff explains well why we're doing this.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -253,6 +253,13 @@ jobs:
       - run: cargo clippy --version
       - run: bash automation/all_clippy_checks.sh
       - save-sccache-cache
+  Check Protobuf files are up-to-date:
+    docker:
+      - image: circleci/rust:latest
+    resource_class: small
+    steps:
+      - checkout
+      - run: ./automation/check_protobuf_files_current.py
```
Comment (thomcc) [inline .circleci/config.yml:262]: nemesis...

File: components/viaduct/src/lib.rs [hunk 2]
```diff
@@ -19,10 +19,7 @@ pub use headers::{consts as header_names, Header, HeaderName, Headers, InvalidHe
 pub use settings::GLOBAL_SETTINGS;
 
 pub(crate) mod msg_types {
-    include!(concat!(
-        env!("OUT_DIR"),
-        "/mozilla.appservices.httpconfig.protobuf.rs"
-    ));
+    include!("mozilla.appservices.httpconfig.protobuf.rs");
```
Comment (thomcc) [inline components/viaduct/src/lib.rs:22]: any reason not to do these as e.g. `#[path = "mozilla.appservices.httpconfig.protobuf.rs"] pub(crate) mod msg_types;`?

File: components/fxa-client/src/mozilla.appservices.fxaclient.protobuf.rs [hunk 3]
```diff
@@ -0,0 +1,181 @@
+#[derive(Clone, PartialEq, ::prost::Message)]
```
Comment (thomcc) [inline components/fxa-client/src/mozilla.appservices.fxaclient.protobuf.rs:1]: Any chance these can get a `// DO NOT EDIT this file was automatically generated` header on these or similar?

Comment (eoger) [reply components/fxa-client/src/mozilla.appservices.fxaclient.protobuf.rs:1]: Good idea, opened https://github.com/danburkert/prost/pull/300.

File: components/viaduct/src/lib.rs (hunk 2 above)
Comment (eoger) [reply components/viaduct/src/lib.rs:22]: Had no idea that was a thing!

Comment (eoger) [reply components/viaduct/src/lib.rs:22]: Well the problem is that these files then get formatted by `cargo fmt`, and the `ignore` directive is only on the nightly channel... 😐

**Pull request merged**